### PR TITLE
chore(sew): improvments to staking event tracker

### PR DESCRIPTION
**Pull request by Lazar955**
This should solve a long lock wait time, which happens when fetching delegations into the tracker and trying to update a tracker

### CODE REVIEW

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go [hunk 1]
```diff
@@ -100,6 +101,7 @@ func NewStakingEventWatcher(
 		metrics:                 metrics,
 		unbondingTracker:        NewTrackedDelegations(),
 		pendingTracker:          NewTrackedDelegations(),
+		inProgressTracker:       NewTrackedDelegations(),
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:104]: Some doc comment would great about what exactly is `inProgressTracker` delegation ?

File: btcstaking-tracker/stakingeventwatcher/tracked_delegations.go [hunk 2]
```diff
@@ -105,6 +105,28 @@ func (td *TrackedDelegations) AddDelegation(
 	return delegation, nil
 }
 
+func (td *TrackedDelegations) AddDel(
```
Comment (KonradStaniec) [inline btcstaking-tracker/stakingeventwatcher/tracked_delegations.go:108]: hmm we now have to function to add to tracker `AddDelegation` and `AddDel` ? How should the caller know which function to use ? It would be great to:
- either have some docs about when each methods should be used
- or have only one methods to add to trackers

**Review by KonradStaniec — APPROVED** [review]
2 non-blocking comments.

Tbh it is getting a bit hard to follow what is happening inside the event tracker, and when each of the `TrackedDelegations` objects  should be used.

File: btcstaking-tracker/stakingeventwatcher/tracked_delegations.go (hunk 2 above)
Comment (Lazar955) [reply btcstaking-tracker/stakingeventwatcher/tracked_delegations.go]: Good point, not sure where my head was, removed the AddDel, it's completely unnecessary

File: btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go (hunk 1 above)
Comment (Lazar955) [reply btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go:106]: Agreed, added a doc string. 

inProgressTracker is basically a cache that says, hey we are trying to activate this (in a go routine) so let's not try again, hence the check in the for loop, and removal in the go routine from it.
